src/boot/efi/stub.c

```diff
@@ -912,16 +976,18 @@ static EFI_STATUS run(EFI_HANDLE image) {
         generate_embedded_initrds(loaded_image, sections, initrds);
         lookup_embedded_initrds(loaded_image, sections, initrds);
 
+        /* Measures ucode addons and populates all_initrds with the final order of all initrds */
+        measure_ucode_addons_and_create_final_list(initrds, &all_initrds, &n_all_initrds, ucode_addons, n_ucode_addons, &sections_measured);
+
         /* Export variables indicating what we measured */
         export_pcr_variables(sections_measured, parameters_measured, sysext_measured, confext_measured);
 
         /* Combine the initrds into one */
         _cleanup_pages_ Pages initrd_pages = {};
         struct iovec final_initrd;
-        if (initrds_need_combine(initrds)) {
-                /* If we have generated initrds dynamically or there is a microcode initrd, combine them with
-                 * the built-in initrd. */
-                err = combine_initrds(initrds, _INITRD_MAX, &initrd_pages, &final_initrd.iov_len);
+        if (n_all_initrds > 1) {
```

Comment:
i think it would make sense to change the line `final_initrd = initrds[INITRD_BASE]` line in the `else` branch that belongs to this `if` to `final_initrd = all_initrds[0]` here. Would of course not actually make any difference, but I think it makes things clearer.